nvflare/private/fed/app/simulator/simulator_runner.py

```diff
@@ -467,11 +530,14 @@ def do_one_task(self, client, num_of_threads, gpu, lock):
 
     def _create_connection(self, open_port):
         conn = None
+        start = time.time()
         while not conn:
             try:
                 address = ("localhost", open_port)
                 conn = Client(address, authkey=CommunicationMetaData.CHILD_PASSWORD.encode())
             except BaseException:
+                if time.time() - start > 60.0:
+                    raise RuntimeError(f"Failed to create connection to the child process in {self.__class__.__name__}")
```

Comment:
We might want to make this 60 seconds configurable.

And we should include this 60 seconds timeout in the `RuntimeError`?

`f"Failed to create connection to the child process in {self.__class__.__name__} in {timeout} seconds"`

Reply:
Done.